ONSdigital/respondent-home-ui
# Sus022 google analytics

# Motivation and Context
Social business need ability to track events from respondent home to monitor performance and issues.
Added google analytics tracking events to RH form submit button and error messages.

# What has changed
- Set up RH properties in Google analytics (for pre-prod and prod - see https://github.com/ONSdigital/ras-deploy/pull/89)
- Store universal analytics IDs as environment variable passed to templates
- Added pattern library analytics event tracking attributes to submit button
- Added pattern library analytics script to base template
- Categorised all RH error message in types to enable ga tracking labels
- Added pattern library analytics event tracking attributes to error messages
- Added pattern library analytics event tracking attributes to system error pages
- Unit tests

# How to test?
- Check GA script is being loaded
- Check analytics events for RH form submit button, for each error message type, and the 500 error page are being tracked in google analytics (request access to GA account for RH properties via [Alison Hawkesworth](mailto:alison.hawkesworth@ons.gov.uk))

# Links
https://trello.com/c/HGbv8xp0/267-sus022-google-analytics-8
